internal/connector/internal/services/connector_types.go

```diff
@@ -350,3 +352,30 @@ func (cts *connectorTypesService) DeleteUnusedAndNotInCatalog() *errors.ServiceE
 	glog.V(5).Infof("Deleted Connector Type with id NOT IN: %v", notToBeDeletedIDs)
 	return nil
 }
+
+func (cts *connectorTypesService) ListCatalogEntries(listArgs *coreService.ListArguments) ([]config.ConnectorCatalogEntry, *api.PagingMeta, *errors.ServiceError) {
+
+	pagingMeta := api.PagingMeta{
+		Page:  listArgs.Page,
+		Size:  listArgs.Size,
+		Total: len(cts.connectorsConfig.CatalogEntries),
+	}
+
+	begin := (pagingMeta.Page - 1) * pagingMeta.Size
+	if begin >= pagingMeta.Total {
+		return []config.ConnectorCatalogEntry{}, &pagingMeta, nil
+	}
+
+	end := begin + pagingMeta.Size
+	if end >= pagingMeta.Total {
+		end = pagingMeta.Total
+	}
+
+	if pagingMeta.Size > pagingMeta.Total {
+		pagingMeta.Size = pagingMeta.Total
+	}
+
+	// TODO it may be better to copy the entries to avoid potential changes outside the control of the service
+	// TODO it may be better to load it from the db ?
+	return cts.connectorsConfig.CatalogEntries[begin:end], &pagingMeta, nil
```

Comment:
@valdar @dhirajsb this is likely not to be the perfect solution but the it is mainly aimed to get information about the deployed catalogs hence it does not rely on the information in the database. 

Eventually we could add an additional endpoint to check for inconsistencies and/or include some stats in the metrics.

Reply:
We should load it from the DB, as it's theoretically possible that there are used older types no longer in the catalog.

Reply:
I think we need to reason a little bit about that as if a connector type is not more in the catalog, then it should not be possible to create a connector from that connector_type hence, it should not be listed among the connector_types 

I don't think that as today we handle such case so theoretically one would be able to create connectors on something we have removed as long an instance of such connector exists which does not seems right. 

I think it could make sense to have an endpoint to get only the shard meta which is where it still make sense to keep older version around.

What do you think ?

Reply:
I agree about the edge case with obsolete connector types. That's the way we are handling the import right now, but perhaps we can delete it and mark older connectors as not having a spec anymore, so one could only delete such connectors. Can you create another issue for that?